### DISCUSSION TITLE
Currydouble UMI002 Curtain Switch No Touch

### DISCUSSION THREAD

**Discussion by mediarath** [arendst/Tasmota#12634] [Q&A] [answered]
Hi guys,

I flashed the Currydouble Curtain Switch today according to the instructions on: [https://templates.blakadder.com/currydouble_UMI002.html](url)
It is the 4Gen with the ESP8266-S1 chip.

Unfortunately the touch buttons do not work. I first thought maybe they were defective. So I tried a second one before and then flashed it. Same problem. How can I change the template so that I can also operate this via touch? In the WebGui everything works fine.
I have tried the Tasmota 9.5 and 9.1. But same effect. I think that the template of Blackadder must be updated. 

Thanks in advance for your help.

**Comment by sfromis** [comment, marked as answer]
Looks like you'll have to build a binary to include support for FTC532, to match the rules suggested for the device.

**Reply by mediarath** [reply]
Great! Thank you. That's it. Now everything works as it should.

**Reply by 1-am-r00t** [reply]
how come the support for FTC532 is not added by default?
I'm struggling with the same issue like OP did, however, I have no clue how to compile the binary myself (tasmota-lite + added FTC532 support). Do I just add features listed in BUILDS.md from the "lite" column to user_config_override.h and then compile it using GitPod?
Willing to learn, so any help greatly appreciated!

**Reply by ascillato** [reply]
For compiling you need to start from Tasmota standard version. Tasmota-lite won't allow you to enable/disable features.
See https://tasmota.github.io/docs/Compile-your-build/

If you need further assistance, please, address this to the [Tasmota Support Chat](https://discord.gg/Ks2Kzd4). The chat is a better and more dynamic channel for helping you. In Discord, there is an specific channel to talk about compiling. Also, you can search in older issues where this has been already addressed.

Thanks.

---------------------------

### Support Information 
See [Docs](https://tasmota.github.io/docs/) for more information.
See [Chat](https://discord.gg/Ks2Kzd4) for more user experience.
See [Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)

**Reply by 1-am-r00t** [reply]
Thanks, but I just managed to compile it on my own. For documentation purposes, I'll briefly describe what I did, in case someone stops by who isn't too interested in using Discord (like myself) or non-archived chat histories:

- Open Tasmocompiler in GitPod (you can register on the fly using your GitHub account: https://gitpod.io/#https://github.com/benzino77/tasmocompiler) and follow instructions on the page
- Select features as needed (in my case ESP8266 Generic, Display Vcc, Mqtt over TLS, Rules, Timers, Tuya MCU, Web Interface)
- Custom parameters for FTC532 support:
`#ifndef USE_FTC532`
` #define USE_FTC532`
`#endif`
- Select version as desired (in my case v9.5.0, english) and compile
- Meanwhile, download https://ota.tasmota.com/tasmota/release/tasmota-minimal.bin
- When compiling is done, download the FIRMWARE.BIN.GZ
- go to Tasmota Web-UI of the Switch and head to Firmware Upgrade and browse for the "tasmota-minimal.bin" file and flash it (https://tasmota.github.io/docs/Upgrading/#upload-buffer-miscompare-error)
- When you're greeted with "MINIMAL Firmware please upgrade", go again to Firmware Upgrade but this time select the "firmware.bin.gz"
- if everything goes right, you now have a functioning touch-switch :-)
